.github/workflows/build-and-push-image-development.yml

```diff
@@ -5,46 +5,94 @@ on:
     branches:
       - main
 
+env:
+  DOCKER_IMAGE: acatran-app
+
 jobs:
-  build-and-push-image-development:
-    name: Build and push image development
+  set-env:
+    name: Prepare
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.var.outputs.branch }}
+      release: ${{ steps.var.outputs.release }}
+    steps:
+      - id: var
+        run: |
+          GIT_REF=${{ github.ref }}
+          GIT_BRANCH=${GIT_REF##*/}
+          RELEASE=dev-`date +%Y-%m-%d`.${{ github.run_number }}
+          echo "branch=$GIT_BRANCH" >> $GITHUB_OUTPUT
+          echo "release=${RELEASE}" >> $GITHUB_OUTPUT
+
+  build-and-push-image:
+    name: Build and push image
+    needs: [ set-env ]
     environment: Dev
     runs-on: ubuntu-latest
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
 
-      - name: Azure Container Registry login
+      - name: Login to ACR
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DEVELOPMENT_AZURE_ACR_CLIENTID }}
           password: ${{ secrets.DEVELOPMENT_AZURE_ACR_SECRET }}
           registry: ${{ secrets.DEVELOPMENT_AZURE_ACR_URL }}
 
-      - name: Prepare tags
-        id: prepare-tags
-        run: |
-          DOCKER_IMAGE=${{ secrets.DEVELOPMENT_AZURE_ACR_URL }}/acatran-app
-          VERSION=dev-`date +%Y-%m-%d`.${{ github.run_number }}
-          SHA=sha-${GITHUB_SHA}
-          TAGS="${DOCKER_IMAGE}:${VERSION},${DOCKER_IMAGE}:${SHA}"
-          echo "tags=${TAGS}" >> $GITHUB_OUTPUT
-          echo "deploy-version=${VERSION}" >> $GITHUB_OUTPUT
-
-      - name: Push image
+      - name: Build and push image
         uses: docker/build-push-action@v3
         with:
           context: .
-          file: ./Dockerfile.gpaas-azure-migration
+          file: Dockerfile.gpaas-azure-migration
+          tags: |
+            ${{ secrets.DEVELOPMENT_AZURE_ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ needs.set-env.outputs.branch }}
+            ${{ secrets.DEVELOPMENT_AZURE_ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ needs.set-env.outputs.release }}
+            ${{ secrets.DEVELOPMENT_AZURE_ACR_URL }}/${{ env.DOCKER_IMAGE }}:latest
           push: true
-          tags: ${{ steps.prepare-tags.outputs.tags }}
 
-      - name: Azure login with ACA credentials
+  create-tag:
+    name: Tag and release
+    needs: [ set-env ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Create tag
+        run: |
+          git tag ${{ needs.set-env.outputs.release }}
+          git push origin ${{ needs.set-env.outputs.release }}
+
+      - name: Create release
+        uses: "actions/github-script@v6"
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          script: |
+            try {
+              await github.rest.repos.createRelease({
```

Comment:
Just for my understanding is this node? It looks like js.
Please could you add a link to the documentation in the comments, just in case we need to make any changes later.

Reply:
It's Javascript, and it's utilising the GitHub API, which is contextually scoped by the [`actions/github-script`](https://github.com/actions/github-script#actionsgithub-script)